ibis/bigquery/client.py

```diff
@@ -305,24 +348,30 @@ def _build_ast(self, expr, context):
         return result
 
     def _execute_query(self, dml, async=False):
+        if async:
+            raise NotImplementedError(
+                'Asynchronous queries not implemented in the BigQuery backend'
+            )
         klass = self.async_query if async else self.sync_query
         inst = klass(self, dml, query_parameters=dml.context.params)
         df = inst.execute()
         return df
 
     def _fully_qualified_name(self, name, database):
         dataset_id = database or self.dataset_id
-        return dataset_id + '.' + name
+        return '{}.{}.{}'.format(self.data_project_id, dataset_id, name)
 
     def _get_table_schema(self, qualified_name):
-        return self.get_schema(qualified_name)
+        _, dataset, table = qualified_name.split('.')
```

Comment:
Why discard the project?

Reply:
I'm assuming (possible incorrectly) that `self.client`'s project is the one we're getting data from.